src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -63,7 +63,30 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
             return DbConnectionValidatorResult.InvalidConnection;
         }
 
-        var factory = GetFactory(providerName, connectionString);
+        IConnectionFactory factory = null;
```

Comment:
Can you extract this into a method just so the `ValidateAsync` isn't a 100 lines long.